### ISSUE TITLE
`marquee` and `timer` listed as live region roles but have intrinsic `aria-live="off"`

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/aria#2144]
In https://www.w3.org/TR/wai-aria-1.3/#live_region_roles `marquee` and `timer` are listed as live region roles. However, when then following through to their definitions, it states:

> Elements with the role `marquee` have an implicit `aria-live` value of `off`.

https://www.w3.org/TR/wai-aria-1.3/#marquee

> Elements with the role `timer` have an implicit `aria-live` value of `off`.

https://www.w3.org/TR/wai-aria-1.3/#timer

As `aria-live="off"` is the default for all elements, isn't this contradictory? I'd have expected that roles listed as being live regions would have an implicit value of either `polite` or `assertive`.

Unless there's some extra "magic" here that is not explained - "they're `off` but still behave differently in some unspecified way compared to other elements"? But then that aspect should be clarified explicitly... looking at https://www.w3.org/TR/wai-aria-1.3/#aria-live there's a lot of prose explaining `assertive` and `polite`, but for `off` you only get the short definition in the table 

> Indicates that updates to the region should not be presented to the user unless the user is currently focused on that region.

But again, if this is the *default*, I would not expect a difference in an element that either has no `aria-live` attribute at all, and one that has `aria-live="off"`.

**Comment by JAWS-test** [comment]
As it is currently defined, it does not make sense. However, the idea behind this could have been: there are live regions whose automatic output by the screen reader would generally be disruptive and unnecessary (as would indeed be the case with marquee and timer), but which could be activated permanently or temporarily by a specific screen reader command. I.e. if I want to know how much time I have left until logout, I press INS+[key for live regions] and hear the last updated value in live regions with aria-live=off once. If I press INS+CTRL+[Key for live regions], I activate the output of aria-live=off permanently.

**Comment by JAWS-test** [comment]
Another reason for the role could be that although I do not get the changes automatically, when I reach the element with the screen reader, I hear the element role and thus know that something changes here from time to time - a reference to areas with dynamic changes and the type of change, so to speak

**Comment by patrickhlauke** [comment]
>  the idea behind this could have been

again, while I appreciate your thoughts on this @JAWS-test ... the idea of me filing the issue here is not to speculate ("what may the spec writers have meant by this?"), but to ask them directly...

**Comment by JAWS-test** [comment]
@patrickhlauke My answer is not intended to question your ticket (which I support: "As it is currently defined, it does not make sense"), but to consider how the ticket can be resolved. The roles can be [deprecated](https://github.com/w3c/aria/issues/1104) or better specified or a difference between a new aria-live=silent and aria-live=off can be introduced

**Comment by spectranaut** [comment]
Discussed today during new issue triage: https://github.com/w3c/aria/issues/2144

**Comment by patrickhlauke** [comment]
@spectranaut was that meant to link to the actual discussion? currently just points back to itself here, and not sure how https://github.com/w3c/core-aam/issues/224 that also references this issue here relates to it /cc @scottaohara

**Comment by scottaohara** [comment]
i think that was just a mistake copy/paste.  short summary is that there is some work to clarify expectations here.  here are the minutes, but it was just discussed at a high level in the largely issue triage meeting we had today - https://www.w3.org/2024/03/14-aria-minutes.html

**Comment by MarioBatusic** [comment]
The value "off" is really nonsensical. Maybe we could introduce the value "silent" and use it as the default for theese roles. So we had in the characteristics table of "aria-live" instead for "off" the description for "silent" stated identically as now for "off":

> Indicates that updates to the region should not be presented to the user unless the user is currently focused on that region.

Additionally, screen readers could offer an option to mute / unmute silent live regions, so the user had not to search for them in the page.

**Comment by patrickhlauke** [comment]
if the idea indeed is that "absence of `aria-live` attribute`" and "`aria-live="off"`" are two distinct separate things (?), then agree that "`off`" is a very misleading/confusing value.

if the idea is that `aria-live="off"` does indeed mean "only announced when the user is focused on that region", then `silent` doesn't sound that much clearer either. without wanting to bike-shed too much, perhaps more `aria-live="focused"` or similar may better convey what is actually meant to go on (assuming I understood the intended nuances right here).

**Comment by smhigley** [comment]
#1104 has some relevant discussion related to this issue

**Comment by cookiecrook** [comment]
Agreed that `silent` is not clear... `focused` is closer but ignores two things:

1. there's no clear distinction of AT focus vs KB focus
2. there's no allowance in the name for the "watched" scenario... e.g. SR user marks it as "tell me when this changes" and we'd want to make that work as expected in combination with `aria-relevant` and `aria-atomic`

**Comment by patrickhlauke** [comment]
again, at the risk of bikeshedding too much here... `aria-live="local"` ?

also, judging by this discussion here, am I now to understand that my assumption here is indeed correct?

>  "absence of `aria-live` attribute" and "`aria-live="off"`" are two distinct separate things

and if so, this definitely needs to be clearly documented (and then separately come up with a better name, and further clarification for, whatever `off` may be/become)

**Comment by pkra** [comment]
Discussed on today's call https://www.w3.org/2024/03/28-aria-minutes.html#t06

**Comment by patrickhlauke** [comment]
From the minutes

> mattking: do we need to define undefined?
>
> jamesn: I'm not sure.
> … spec says "off" is the default.

and this, as well, is likely part of the confusion for developers. saying some value is the default seems to indicate "so even if you leave out the attribute, that's the assumed default" ... but i assume the "default" here is more "if you just add `aria-live` with no value, it defaults to `"off"`)

**Comment by cookiecrook** [comment]
Test case here. https://bug-271841-attachments.webkit.org/attachment.cgi?id=470641

**Comment by cookiecrook** [comment]
@patrickhlauke wrote:
> separately come up with a better name

yeah, discussion was that "off" is too confusing for authors given the original intention... spec needs to add 1. an `undefined` default, and 2. probably a more clear synonym for "off" (perhaps "local" as patrick and others suggested, but it's a fairly technical name that may still be unclear to some authors)

**Comment by spectranaut** [comment]
Discussed briefly in yesterday's meeting: https://www.w3.org/2024/04/04-aria-minutes#t08

**Comment by patrickhlauke** [comment]
https://www.w3.org/2024/04/04-aria-minutes#t06

**Comment by patrickhlauke** [comment]
dropping this here just for reference, but ... a recent article making the rounds that also misunderstands what `aria-live="off"` means https://scribe.rip/@askParamSingh/exploring-the-power-of-aria-live-attribute-in-web-accessibility-f80e6d110ed4#b1a2

> aria-live=”off”: By default, it’s set to off. This value indicates that changes to the content within the region should not be announced by assistive technologies.

**Comment by cookiecrook** [comment]
cc @rahimabdi for the IDL context

**Comment by cookiecrook** [comment]
I have a proposed solution in this related issue:
- https://github.com/w3c/aria/issues/2674#issuecomment-3555965835

And a pull request for ARIA and Core-AAM:
- #2677
